mixing/mixpool/observer.go

```diff
@@ -0,0 +1,503 @@
+// Copyright (c) 2025 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package mixpool
+
+import (
+	"context"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/decred/dcrd/chaincfg/chainhash"
+	"github.com/decred/dcrd/wire"
+)
+
+const strikeLimit = 2
+
+const minPeers = 4 // Keep synced with mixclient.MinPeers
```

Comment:
Maybe it should be exported from mixpool and used by mixclient since mixclient already depends on mixpool?